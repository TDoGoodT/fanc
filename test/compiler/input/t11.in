
int decrement(int n) {
    return n - 1;
}
int increment(int n) {
    return n + 1;
}
int multiply(int a, int abb) {
    return a * abb;
}

int power(int base, int exponent) {
    int result = 1;
    int count = 0;
    while (count < exponent) {
        result = multiply(result, base);
        count = increment(count);
    }
    return result;
}

void main() {
    int base = 2;
    int exponent = 5;

    int result = power(base, exponent);

    print("2 raised to the power of 5: ");
    printi(result);
}
