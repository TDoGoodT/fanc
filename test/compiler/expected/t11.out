2 raised to the power of 5: 
32
